filament/src/components/RenderableManager.cpp

```diff
@@ -301,10 +304,49 @@ RenderableManager::Builder::Result RenderableManager::Builder::build(Engine& eng
 }
 
 RenderableManager::Builder& RenderableManager::Builder::instances(size_t instanceCount) noexcept {
+    ASSERT_PRECONDITION(mImpl->mInstanceTransforms == nullptr && mImpl->mInstanceBuffer == nullptr,
+            "cannot call instances(size_t) override, instance local transforms already provided");
     mImpl->mInstanceCount = clamp((unsigned int)instanceCount, 1u, 32767u);
     return *this;
 }
 
+RenderableManager::Builder& RenderableManager::Builder::instances(size_t instanceCount,
+        math::mat4f const* localTransforms) noexcept {
+    ASSERT_PRECONDITION(instanceCount >= 1, "instanceCount must be >= 1.");
```

Comment:
typically, we do all the validation in `build`, not in the individual builder methods.

the idea is that there is a single method that "is heavy" and "can fail".